lib/ecto/adapters/postgres/connection.ex

```diff
@@ -1619,26 +1619,24 @@ if Code.ensure_loaded?(Postgrex) do
     end
 
     defp column_type(:identity, opts) do
-      start_value = [Keyword.get(opts, :start_value)]
-      increment = [Keyword.get(opts, :increment)]
+      start_value = Keyword.get(opts, :start_value)
+      increment = Keyword.get(opts, :increment)
+      cache = Keyword.get(opts, :cache)
+      clause = Keyword.get(opts, :clause, :by_default) |> to_string |> String.upcase |> String.split("_") |> Enum.join(" ")
       type_name = ecto_to_db(:identity)
 
-      cleanup = fn v -> is_integer(v) and v > 0 end
 
-      sequence =
-        start_value
-        |> Enum.filter(cleanup)
-        |> Enum.map(&"START WITH #{&1}")
-        |> Kernel.++(
-          increment
-          |> Enum.filter(cleanup)
-          |> Enum.map(&"INCREMENT BY #{&1}")
-        )
+      [start_value]
+        |> Enum.filter(&(is_integer/1)) |> Enum.map(&"START WITH #{&1}")
+        |> Kernel.++( [increment] |> Enum.filter(&(is_integer/1)) |> Enum.map(&"INCREMENT BY #{&1}") )
+        |> Kernel.++( [cache] |> Enum.filter(&(is_integer(&1) && Kernel.>(&1, 0))) |> Enum.map(&"CACHE #{&1}"))
+        |> case do
+          [] -> []
+          [h | t] -> [["(", h], [t, ") "]]
+        end
+        |> (&([type_name, " GENERATED #{clause} AS IDENTITY", &1 |> Enum.join(" ")])).()
```

Comment:
Please assign the result to a variable, instead of making everything fit a pipeline.

Reply:
Haha was pushing hard to see how far I can take the chain. The original one was using then/2 as well.